src/modules/m_denychans.cpp

```diff
@@ -120,9 +120,13 @@ class ModuleDenyChannels : public Module
 				continue;
 
 			// If the redirect channel is whitelisted then it is okay.
-			for (GoodChannels::const_iterator j = goodchans.begin(); j != goodchans.end(); ++j)
+			bool whitelisted = false;
+			for (GoodChannels::const_iterator j = goodchans.begin(); j != goodchans.end() && !whitelisted; ++j)
```

Comment:
This would be clearer if you used break inside the if statement if something is whitelisted.